### DISCUSSION TITLE
Longitud maxima de linea

### DISCUSSION THREAD

**Discussion by agmblanco** [dds-jv/foro#420] [Consultas de herramientas] [answered]
Buenas tardes, cómo están? Quería saber si tenemos permitido modificar el tamaño máximo de línea en el estilo, 100 caracteres. En mi opinión y la de otros con los que tuve oportunidad de hablar del tema, resulta un número demasiado bajo. Más teniendo en cuenta que, por un lado, Java es un lenguaje particularmente verboso. Y, por el otro, es esperable, por tratarse de la relación de aspecto más común, que quien vaya a leer el código lo haga en un monitor de 16:9, por lo que no encuentro el sentido a limitar el tamaño de las líneas a un formato que fue pensado para ser fácilmente imprimible y desperdicia la mayor parte de espacio en pantalla, obligándote a leer hacia abajo en lugar de hacia el costado, cómo es habitual en nuestro lenguaje.

**Comment by flbulgarelli** [comment, marked as answer]
:wave: ¡Hola!

¿Estamos hablando del TPI o del TPA? Si estamos hablando del TPI, no hay problema, dado que serás la única persona que edite ese código. 

Ahora bien, si estamos hablando del TPA, la cosa es distinta, porque deberías pensar que esos límites tienen que ser consensuados con tu equipo (y no todo el mundo en el mismo tendrá necesariamente la misma capacidad visual, o monitores 16:9, o no necesariamente grandes, o incluso habrá personas que consulten ocasionalmente el código desde sus celulares). Es más, no todas las personas utilizarán modo de pantalla completa, y puede que tengan uno o mas componentes laterales.  

Sin ir mas lejos, en lo personal en el monitor que suelo utilizar, con el editor, tipografía, tamaño de fuente que me resulta cómoda, ya con una barra lateral abierta, el ancho neto que tengo disponible es de 101 a 105 caracteres (~125 con barras laterales cerradas, pero no es un caso de uso muy realista), no muy diferente a los 100 que plantea el formateador que sugerimos, que as su vez, está basado en el formateador de Google: 

https://github.com/checkstyle/checkstyle/blob/e6051f20c20511f8d44913b3cf6eb5fb113f2343/src/main/resources/google_checks.xml#L45-L49

> Quería saber si tenemos permitido modificar el tamaño máximo de línea en el estilo, 100 caracteres.

¿A cuánto querrías modificarlo? El estándar de Java es de 80: https://www.oracle.com/java/technologies/javase/codeconventions-indentation.html. Y el de Apache, 120: https://bval.apache.org/coding-standards.html. Luego, en general encontrarás que los proyectos suelen adherir a algo en ese rango. Si todo el equipo (incluyendo ayudantes) está de acuerdo, adelante nomás, pero no me iría de esos parámetros.

**Reply by flbulgarelli** [reply]
Ah, PD: me doy cuenta ahora de que estás en mi equipo, así que no, para el TPA dejalo en 100 por lo que te mencionaba antes :smile:

**Comment by agmblanco** [comment]
Hola Fran, originalmente hablaba de ambos, pero entiendo el punto. Gracias por la respuesta. Saludos!
